src/persist-client/src/read.rs

```diff
@@ -752,7 +752,14 @@ where
                 .machine
                 .heartbeat_leased_reader(&self.reader_id, heartbeat_ts)
                 .await;
-            if !existed {
+            if !existed && !self.machine.state().collections.is_tombstone() {
+                // It's probably surprising to the caller that the shard
+                // becoming a tombstone expired this writer. Possibly the right
```

Comment:
fixed this "writer" -> "reader" and the "WriterId" -> "LeasedReaderId" below, but don't want to interrupt my CI runs, so leaving it on my local branch for now